services/notification/notifiers/codecov_slack_app.py

```diff
@@ -84,6 +84,7 @@ def build_payload(self, comparison: ComparisonProxy) -> dict:
             message = "unknown"
             notation = ""
             comparison_url = None
+        head_report = comparison.head.report
```

Comment:
Is `comparison.head` typed as a `FullCommit` in the code? Maybe it would be a good idea to update the type to account for that. I've seen examples where the `commit` might also be `None` and have a half-done PR to update it

Reply:
I’m unsure about that one. I had the problem that `comparison` might be wrongly typed, as some tests were failing because it was actually a `FilteredComparison`, though I’m not sure if that only happened because of tests, or if that can also happen in real world code. In other parts of the notification infra, I have updated the types to take that into account.